package com.google.common.hash does not exist when building keycloak-api-docs-dist

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

docs

### Describe the bug

When building `keycloak-api-docs-dist` there is the following error:

```
Error:  Constructing Javadoc information...
Error:  /home/runner/work/keycloak-rel/keycloak-rel/server-spi-private/src/main/java/org/keycloak/policy/BlacklistPasswordPolicyProviderFactory.java:20: error: package com.google.common.hash does not exist
Error:  import com.google.common.hash.BloomFilter;
Error:                               ^
Error:  /home/runner/work/keycloak-rel/keycloak-rel/server-spi-private/src/main/java/org/keycloak/policy/BlacklistPasswordPolicyProviderFactory.java:21: error: package com.google.common.hash does not exist
Error:  import com.google.common.hash.Funnels;
Error:                               ^
Error:  /home/runner/work/keycloak-rel/keycloak-rel/server-spi-private/src/main/java/org/keycloak/policy/BlacklistPasswordPolicyProviderFactory.java:249: error: cannot find symbol
Error:          private BloomFilter<String> blacklist;
Error:                  ^
Error:    symbol:   class BloomFilter
Error:    location: class FileBasedPasswordBlacklist
Error:  /home/runner/work/keycloak-rel/keycloak-rel/server-spi-private/src/main/java/org/keycloak/policy/BlacklistPasswordPolicyProviderFactory.java:303: error: cannot find symbol
Error:          private BloomFilter<String> load() {
Error:                  ^
Error:    symbol:   class BloomFilter
Error:    location: class FileBasedPasswordBlacklist
Error:  /home/runner/work/keycloak-rel/keycloak-rel/server-spi-private/src/main/java/org/keycloak/policy/BlacklistPasswordPolicyProviderFactory.java:328: error: cannot find symbol
Error:          protected void insertPasswordsInto(BloomFilter<String> filter) throws IOException {
Error:                                             ^
Error:    symbol:   class BloomFilter
Error:    location: class FileBasedPasswordBlacklist
Error:  warning: unknown enum constant Scopes.NAMED_CACHE
Error:    reason: class file for org.infinispan.factories.scopes.Scopes not found
```

See the whole log here: https://github.com/keycloak-rel/keycloak-rel/actions/runs/9200474260/job/25307118255

This is caused #28385 by removing the following lines:
https://github.com/keycloak/keycloak/commit/f32cd91792f434754640d0213c65d7faf5e101a8#diff-f3cb79545f6f970311565315ed40d17960acf7cd71760259ff2642bb50cbdb21L66-L71


### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

```
./mvnw clean install -f distribution/api-docs-dist 
```

### Anything else?

_No response_

--- a/distribution/api-docs-dist/pom.xml
+++ b/distribution/api-docs-dist/pom.xml
@@ -63,6 +63,11 @@
             <groupId>net.java.dev.jna</groupId>
             <artifactId>jna</artifactId>
         </dependency>
+        <!-- This is here because BloomFilter is used in the Javadoc within BlacklistPasswordPolicyProviderFactory -->
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+        </dependency>
     </dependencies>
 
     <build>

